### PR TITLE
Integration ONNX Editor with FE API

### PR DESCRIPTION
**Pull request by mbencer**
Tikcet: 57250

### CODE REVIEW

File: ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+        onnx.save_model(model, name)
+        test_models_names.append(name)
+
+
+def teardown_module():
+    for name in test_models_names:
+        os.remove(name)
+
+
+def skip_if_onnx_frontend_is_disabled():
```
Comment (nosovmik) [inline ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py:183]: These tests are currently not executed on CI, right?

File: ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+        make_tensor_value_info("out3", onnx.TensorProto.FLOAT, (4, 6)),
+        make_tensor_value_info("out4", onnx.TensorProto.FLOAT, (8, 16)),
+    ]
+    graph = make_graph([add, split, relu, mul], "test_graph", input_tensors, output_tensors)
+    models["test_partial_shape.onnx"] = make_model(graph, producer_name="ONNX Importer")
+
+    return models
+
+
+fem = FrontEndManager()
```
Comment (nosovmik) [inline ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py:167]: Could you consider to reuse 'C++' tests for testing C++ frontends, like we did for 'PaddlePaddle'? There we generate necessary test models on build time, and then converting those models during testing? Starting from today, these tests are also executed on openvino-lin CI

File: ngraph/frontend/frontend_manager/src/utils.hpp [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 #include <string>
 #include "frontend_manager/frontend_manager_defs.hpp"
+#include "ngraph/partial_shape.hpp"
```
[the remark this answers is not in the document]
Comment (mbencer) [reply ngraph/frontend/frontend_manager/src/utils.hpp:7]: `to_ng_shape` needs it

File: ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py (hunk 1 above)
Comment (mbencer) [reply ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py:183]: The tests are run if `DNGRAPH_ONNX_FRONTEND_ENABLE == TRUE`.
They are executed in `openvino-ngraph-onnx` CI.

File: ngraph/frontend/frontend_manager/src/utils.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (nosovmik) [reply ngraph/frontend/frontend_manager/src/utils.hpp:7]: But why it is needed in utils.hpp? I'd suggest to add `#include` in files which depend on 'to_ng_shape'. This file has only `get_frontend_library_path` and depends on FRONTEND_API macro (in frontend_manager_defs.hpp) and std::string definition (inside <string>). Nothing else is needed for this file

File: ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py (hunk 2 above)
Comment (mbencer) [reply ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py:167]: yes, it seems to be a good idea, but please let us do it separately. I created a ticket to handle it: 60998

File: ngraph/frontend/frontend_manager/src/utils.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbencer) [reply ngraph/frontend/frontend_manager/src/utils.hpp:7]: Right ;) I've mistaken `ngraph/frontend/frontend_manager/src/utils.hpp` with `ngraph/frontend/onnx/onnx_common/include/onnx_common/utils.hpp` where there is declaration of `to_ng_shape`.
Removed.

File: ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+
+def skip_if_onnx_frontend_is_disabled():
+    front_ends = fem.get_available_front_ends()
+    if "onnx" not in front_ends:
+        pytest.skip()
+
+
+# Function to compare ng Functions (ops names, types and shapes).
+# Note that the functions uses get_ordered_ops, so the topological order of ops should be also preserved.
+def comare_functions(current, expected):  # noqa: C901 the function is too complex
```
[the remark this answers is not in the document]
Comment (mbencer) [reply ngraph/python/tests/test_frontend/test_frontend_onnx_editor.py:191]: done

**Review by tomdol — APPROVED** [review]

**Review by ilyachur — COMMENTED** [review]
Can we merge ONNX frontend and ONNX importer? Looks like the purposes for these components are the same.

**Comment by mbencer** [comment]
> Can we merge ONNX frontend and ONNX importer? Looks like the purposes for these components are the same.

Yes, we have plans to do it, but separately, it will be part of 60948

**Review by postrational — APPROVED** [review]

**Review by mateusztabaka — APPROVED** [review]

**Comment by ilyachur** [comment]
@postrational @tomdol @mateusztabaka Why was this PR merged without approve from @nosovmik or @slyalin at least?

**Pull request merged**